codex/contracts/interactions/clientinteractions.nim

```diff
@@ -0,0 +1,55 @@
+import pkg/ethers
+import pkg/chronicles
+
+import ../../purchasing
+import ../deployment
+import ../marketplace
+import ../market
+import ../proofs
+import ../clock
+import ./interactions
+
+export purchasing
+export chronicles
+
+type
+  ClientInteractions* = ref object of ContractInteractions
+    purchasing*: Purchasing
+
+proc new*(_: type ClientInteractions,
+          signer: Signer,
+          deployment: Deployment): ?ClientInteractions =
+
+  without address =? deployment.address(Marketplace):
```

Comment:
Would be nice if we could remove this remaining duplication between `ClientInteractions.new()` and `HostInteractions.new()`. Perhaps introduce a `ContractInteractions.init()` that is called from both of them?

Reply:
What is left that is duplicated? The main difference between the two is that `HostInteractions` takes a RepoStore in its constructor while `ClientInteractions` does not.

Reply:
It's not a biggie, but this code seems to be the same across `HostInteractions` and `ClientInteractions`:

```nim
  without address =? deployment.address(Marketplace):
    let err = newException(ContractAddressError,
      "Unable to determine address of the Marketplace smart contract")
    return failure(err)

  let contract = Marketplace.new(address, signer)
  let market = OnChainMarket.new(contract)
  let clock = OnChainClock.new(signer.provider)
```

Reply:
done